app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -3295,37 +3295,61 @@ class ChatActivity :
         }
     }
 
-    fun shareToNotes(message: ChatMessage, roomToken: String) {
-        var shareUri: Uri? = null
-        val data: HashMap<String?, String?>?
-        var metaData: String = ""
-        var objectId: String = ""
-        if (message.hasFileAttachment()) {
-            val filename = message.selectedIndividualHashMap!!["name"]
-            path = applicationContext.cacheDir.absolutePath + "/" + filename
-            shareUri = FileProvider.getUriForFile(
-                this,
-                BuildConfig.APPLICATION_ID,
-                File(path)
-            )
+    fun shareToNotes(message: ChatMessage) {
+        val apiVersion = ApiUtils.getConversationApiVersion(
+            conversationUser!!,
+            intArrayOf(ApiUtils.API_V4, ApiUtils.API_V3, 1)
+        )
+        chatViewModel.checkForNoteToSelf(
+            ApiUtils.getCredentials(conversationUser!!.username, conversationUser!!.token)!!,
+            ApiUtils.getUrlForRooms(
+                apiVersion,
+                conversationUser!!.baseUrl
+            ),
+            false
+        )
```

Comment:
As long as this is only done when the action is pressed, instead of getting all rooms and searching for the note-to-self, can you GET the room with the token `note-to-self`? That will create it in case it was (accidentally) deleted

Reply:
improved with ab19f631aeb4264e894da4075ce524131f727723